eclipse-collections-code-generator/src/main/resources/impl/bag/immutable/immutablePrimitiveHashBag.stg

```diff
@@ -472,6 +476,14 @@ final class Immutable<name>HashBag implements Immutable<name>Bag, Serializable
             return this.bag.toImmutable();
         }
     }
+    <if(primitive.specializedStream)>
+
+    @Override
+    public Spliterator.Of<name> spliterator()
+    {
+        return Spliterators.spliterator(this.delegate.toArray(), Spliterator.IMMUTABLE);
```

Comment:
This is a very costly implementation of a spliterator. This should not create a copy of the primitive HashBag into an array. It can delegate to the spliterator of the primitive HashBag.

Reply:
Smart - makes sense, thx